src/build.d

```diff
@@ -514,6 +515,29 @@ alias zip = makeDep!((builder, dep) => builder
     })
 );
 
+alias scp = makeDep!((builder, dep) => builder
+    .name("scp")
+    .sources(buildFiles.chain(sources.dmd, sources.lexer,
+        sources.backend, sources.root, sources.frontendHeaders).array)
+    .commandFunction(() {
+        string[][string] sourceDirGroups;
```

Comment:
Using an a AA to join source files accoring to their directory seems unecessary. I would suggest working in batches structured by `sources` members:
```
const batches = [buildFiles, sources.dmd, sources.lexer, sources.backend, sources.root, sources.frontendHeaders];
...
builder.sources(batches.join)
...
foreach(const batch; batches) {
    const dir = env["SCPDIR"].buildPath(source.relativePath(batch[0].dirName);
    // remaining copy code
}
```

Reply:
The current solution is less brittle.  There's nothing preventing source files from different directories being in any one of the sources variables.